### PR TITLE
[Maps] Improve API rate limit errors

### PR DESCRIPTION
**Pull request by jshoe**
### Description
- This adjusts the error handling for the geosearch API. I didn't properly distinguish between API rate limit errors and "0 results" HTTPNotFound errors before.
- For the sake of accurate alerting and monitoring if users ever encounter rate limits.

### Tests
- See tests

### CODE REVIEW

File: app/controllers/locations_controller.rb [hunk 1]
```diff
@@ -18,13 +19,17 @@ def external_search
...
-      if success
+      if success && resp.is_a?(Array)
+        # Successful request with results. Just keep the first if you get duplicate locations.
         results = resp.map { |r| LocationHelper.adapt_location_iq_response(r) }
-        # Just keep the first if you get duplicate locations
         results = results.uniq { |r| [r[:name], r[:geo_level]] }
+      elsif success && resp.is_a?(Hash) && resp["error"] == "Unable to geocode"
+        # Successful request but 0 results
+        Rails.logger.info("No geosearch results for: #{query}")
       else
```
Comment (tfrcarvalho) [inline app/controllers/locations_controller.rb]: Is there no error message from the API? Nothing more specific we can log?

File: app/models/location.rb [hunk 2]
```diff
@@ -46,7 +46,9 @@ def self.location_api_request(endpoint_query)
     resp = Net::HTTP.start(uri.host, uri.port, use_ssl: true) do |http|
       http.request(request)
     end
-    [resp.is_a?(Net::HTTPSuccess), JSON.parse(resp.body)]
+    # Search with 0 results will return HTTPNotFound. Consider it a successful request for our handling.
+    success = resp.is_a?(Net::HTTPSuccess) || resp.is_a?(Net::HTTPNotFound)
```
Comment (tfrcarvalho) [inline app/models/location.rb]: Could HTTPNotFound be an error on our part ? Or if for some reason something changes on our config?

File: app/controllers/locations_controller.rb (hunk 1 above)
Comment (jshoe) [reply app/controllers/locations_controller.rb]: the full response is actually `{ "error": "Unable to geocode" }`  :3

File: app/models/location.rb (hunk 2 above)
Comment (jshoe) [reply app/models/location.rb]: sounds like it's possible yeah. what would you suggest for this?

File: app/controllers/locations_controller.rb (hunk 1 above)
Comment (tfrcarvalho) [reply app/controllers/locations_controller.rb]: I meant if there arent other errors... Can you guarantee that `success && resp.is_a?(Hash) && resp["error"] != "Unable to geocode"` does not happen?

Comment (jshoe) [reply app/controllers/locations_controller.rb]: updated as discussed 👍

File: app/models/location.rb (hunk 2 above)
Comment (tfrcarvalho) [reply app/models/location.rb]: So, with your changes above any other errors will be logged, right?

Comment (jshoe) [reply app/models/location.rb]: right, they will fall through and go to log and Airbrake

**Pull request merged**